### PR TITLE
updating to operator-sdk version 1.31.0

### PR DESCRIPTION
**Pull request by acornett21**
- Updating to operator-sdk version 1.31
- Updating to go 1.20
   - Updating the CI to reference the mod file so we don't have to update for go version changes
- Updating project structure based on migration docs

### CODE REVIEW

**Review by skattoju — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,13 +9,13 @@ jobs:
...
-        go-version: 1.19
-      id: go
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
+    - uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+      id: go
     - id: go-cache-paths
```
Comment (bcrochet) [inline .github/workflows/ci.yml:19]: If I'm not mistaken, I think setup-go@v4 takes care of all the caching now?

File: Makefile [hunk 2]
```diff
@@ -192,35 +191,46 @@ ENVTEST ?= $(LOCALBIN)/setup-envtest
 OPERATOR_SDK ?= $(LOCALBIN)/operator-sdk
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v4.5.4
-CONTROLLER_TOOLS_VERSION ?= v0.10.0
-OPERATOR_SDK_VERSION ?= "v1.26.0"
-
+KUSTOMIZE_VERSION ?= v4.5.7
+CONTROLLER_TOOLS_VERSION ?= v0.11.1
+ OPERATOR_SDK_VERSION ?= "v1.31.0"
```
Comment (bcrochet) [inline Makefile:196]: Extra space at the beginning of line.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Couple of thoughts...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (acornett21) [reply .github/workflows/ci.yml]: v4 does do cacheing by default now, cache steps have been removed.

File: Makefile (hunk 2 above)
Comment (acornett21) [reply Makefile]: good catch.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Comment by acornett21** [comment]
- Fixes: #148

**Pull request merged**